backend/src/main/java/gov/cdc/usds/simplereport/db/model/Facility.java

```diff
@@ -78,35 +80,60 @@ public String getFacilityName() {
 		return facilityName;
 	}
 
+    public DeviceSpecimen getDefaultDeviceSpecimen() {
+        return defaultDeviceSpecimen;
+    }
+
 	public DeviceType getDefaultDeviceType() {
-		return defaultDeviceType;
+        return defaultDeviceSpecimen == null ? null : defaultDeviceSpecimen.getDeviceType();
 	}
 
 	public void setDefaultDeviceType(DeviceType defaultDeviceType) {
-		this.defaultDeviceType = defaultDeviceType;
+        throw new IllegalArgumentException();
 	}
 
 	public List<DeviceType> getDeviceTypes() {
 		// this might be better done on the DB side, but that seems like a recipe for weird behaviors
-		return configuredDevices.stream()
+        return configuredDeviceSpecimens.stream()
 			.filter(e -> !e.isDeleted())
+                .map(DeviceSpecimen::getDeviceType)
+                .filter(e -> !e.isDeleted())
```

Comment:
I don't with that expectation 🙂 

The three different soft-deletion flags represent different logical actions. Soft-deleting a device-specimen means you can't use that specimen type with that device any more, while soft-deleting a device means you can't use that device type any more. (It's sort of hard for me to imagine the case where nobody is allowed to use nasal swabs any more, but what the heck, it's a pretty cheap flag to maintain.) If I undeleted a device type that I had previously deleted, I would expect the previous specimen-type configuration for it to still exist, including respecting any soft-deletions I had previously made of DeviceSpecimens.

Reply:
Two thoughts:
a) Soft-deleting a device should also mean that you can't use *any* specimen type with that device any more, because you can't use that device any more.
b) But if we are in fact supporting undeletion of device types, I don't know how in the world we'd keep track of which deviceSpecimens to un-delete vs which legit bad deviceSpecimens should independently remain deleted, so I'm not going to die on this hill. But if we're not supporting undeletion, I still stick by my gut in (A). Not a hard blocker for the PR though, since this can be added in later if we so choose.

Reply:
c) I'm starting to see this crop up as an issue elsewhere, like https://github.com/CDCgov/prime-simplereport/pull/799#discussion_r580360854 🙁 (at least *i* think it's an issue)